### PR TITLE
[Resolve #1114,#1000] Project Dependencies part 5: Handling resolvers referencing non-deployed stacks in template commands

### PR DESCRIPTION
**Pull request by jfalkenstein**
This is the Fourth in a series of pull requests that addresses #1114 , allowing Sceptre to manage its own dependencies.

## In this PR:
* It is possible to enter a global state (using a context manager) that allows Resolvers that raise errors to resolve to a placeholder value. This is explicitly opt-in and will only be enabled on commands that don't actually deploy stacks, such as "generate", "validate", and "diff". This is required so that these commands don't fail when there are resolvers that reference stacks that aren't deployed yet. This ultimately is the bit that resolves issue #1000. This is important because this series of Pull requests more than doubles the number of stack attributes that are resolvable, which are going to be all the more problematic when using `sceptre validate`, etc...

## PR Checklist

- [x] Wrote a good commit message & description [see guide below].
- [x] Commit message starts with `[Resolve #issue-number]`.
- [x] Added/Updated unit tests.
- [x] Added/Updated integration tests (if applicable).
- [x] All unit tests (`make test`) are passing.
- [x] Used the same coding conventions as the rest of the project.
- [x] The new code passes pre-commit validations (`pre-commit run --all-files`).
- [x] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

**Comment by jfalkenstein** [comment]
@joe1981al This addresses the concern you raised in #1000

File: sceptre/cli/template.py [hunk 1]
```diff
@@ -66,7 +87,11 @@ def generate_command(ctx, path):
     )
 
     plan = SceptrePlan(context)
-    responses = plan.generate()
+
+    execution_context = null_context() if no_placeholders else use_resolver_placeholders_on_error()
```
Comment (zaro0508) [inline sceptre/cli/template.py:91]: it looks like validate and diff contains a more verbose version for this logic.  just curious, any reason why it's different here?

Comment (jfalkenstein) [reply sceptre/cli/template.py:91]: A previous version had parameters passed to use_resolver_placeholders_on_error but then they were removed. I can make them uniform now, though.

**Comment by jfalkenstein** [comment]
I'm going to close this PR in favor of #1185 since I needed to recut that branch properly.

**Pull request closed without merging**